### DISCUSSION TITLE
AC charging not being differentiated from DC

### DISCUSSION THREAD

**Discussion by radioaz** [teslamate-org/teslamate#3196] [Getting started] [answered]
I just installed Teslamate a few days ago, and I love it so far.    Everything is mostly working great.  

However, all of my charging seems to be getting bundled as DC charging, even if it's AC charging at home.
Since installing, I have 2 home charging sessions, and 4 supercharger sessions.   No AC charging is showing up in the Charging Stats graph.   It does say total of 6 charges.

Any thoughts?
Thank you!

**Comment by vzhivkov** [comment]
I also think that it's important to see how much AC and how much DC charging has been done over time.

**Comment by DrMichael** [comment]
Well in Charging Stats it is differentiated by
`WHEN NULLIF(mode() within group (order by charger_phases),0) is null THEN 'DC' ELSE 'AC'
`
So, when phases are 0 it is considered as DC AFAIU.

**Comment by radioaz** [comment]
My one theory why it seems to work for others, and not me, is that I am in North America, and possibly the charging phase data may be different from most users who appear to be in Europe.


Get Outlook for iOS<https://aka.ms/o0ukef>
________________________________

**Comment by 1920x1200** [comment]
Did anyone figure out how to fix this? There are free public chargers at my gym that are definitely AC (the charger shows the voltage, amperage, etc on it's screen) but TeslaMate keeps listing it as DC. I can't find any way to edit this.

**Comment by 1920x1200** [comment]
If this issue can't be prevented/resolved, is there a way I can at least correct the data? For example I can click on the cost and enter in how much I paid pretty easily without having to mess around with the database. Is there a way I can click DC and change it to AC so my stats will be correct?

**Comment by JVital2013** [comment]
I have this issue as well: by default, 100% of charges show as DC. Unless I have a supercharger at my house, this is not likely 😄. Fortunately I found a potential solution. First, some notes:

- My car is a 2013 Model S 60 with the original battery pack (as far as I know, I recently bought it used)
- I have never charged in an area without cell coverage
- `charger_phases` always appears to be `nil` over the API, and is not stored in the database. 
- `fast_charger_present` is always correct.

Below are the results of the query:

```sql
SELECT
    charging_process_id,
    min(date) as date,
    max(charger_power) as charger_power,
    charger_phases,
    fast_charger_present
FROM
    charges
WHERE
    charger_power != 0
GROUP BY
    charging_process_id,
    charger_phases,
    fast_charger_present
ORDER BY
    charging_process_id ASC;
```
<img width="826" height="571" alt="image" src="https://github.com/user-attachments/assets/d32aec3a-c286-416c-92ac-28de4e1b9ce1" />
<br /><br />

**A quick test of editing some of the dashboards look promising:**
<img width="1572" height="727" alt="image" src="https://github.com/user-attachments/assets/e1497601-2e90-4013-a828-90dd4fc5c49f" />
<img width="1573" height="779" alt="image" src="https://github.com/user-attachments/assets/ad8f5e1e-447e-4092-bdfc-ffa1025294af" />


Would it not be better to determine DC charging by the fast_charger_present flag instead of charger_phases? Or is that assumption untrue in some cases I have not seen, such as when using non-Tesla DC chargers?

After all, it works on my machine™ 😄.

**Comment by JakobLichterfeld** [comment, marked as answer]
--> #4401
